src/SearchFilters.js

```diff
@@ -24,7 +24,7 @@ let LabelledList = ({ list, Component }) =>
 export let AddableFilterList = props => (
   <>
     <FilterList {...props} />
-    <FilterAdder {...props} uniqueFields />
+    <FilterAdder {...props} uniqueFields={!props.allowDuplicateFields} />
```

Comment:
I think you also need to thread this through search filters / search tree or whatever, right?